### DISCUSSION TITLE
Isolation not loading cookies

### DISCUSSION THREAD

**Discussion by owly111** [cefsharp/CefSharp#3504] [General]
- **What version of the product are you using?**
    - 89.0.170

- **What architecture x86 or x64?**
    - x64
	
- **What version of .Net?**
    .Net 4.6
    
- **On what operating system?**
    Win7

- **Are you using `WinForms`, `WPF` or `OffScreen`?**
    WinForms
    
- **What steps will reproduce the problem?**
    I am trying to isolate each browser based on the current user, to have more sessions on the same website. This is my code:
`browser = new ChromiumWebBrowser("http://www.whatarecookies.com/cookietest.asp")
            {
                RequestContext = new RequestContext(new RequestContextSettings { CachePath = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData) + @"\FaceNet\user", PersistSessionCookies = true })
            };`

- **What is the expected output? What do you see instead?**
I was expecting to find the old session loaded back, but instead it starts as a completely new browser.

- **Please provide any additional information below.**
    The sessions are being saved in the folder I specified, they just don't get loaded back.

Thank you in advance.

**Comment by amaitland** [comment]
Did you check the log file? 

See https://github.com/cefsharp/CefSharp/issues/3111#issuecomment-629713608

**Comment by owly111** [comment]
> Did you check the log file?
> 
> See [#3111 (comment)](https://github.com/cefsharp/CefSharp/issues/3111#issuecomment-629713608)

I read the link you cited, and by doing some testing I realized my Cookies file is not updating. It gets created the first time, and then stays the same.

**Reply by amaitland** [reply]
What value did you specify for the RootCachePath? It needs to be writable folder by the user.

**Reply by owly111** [reply]
> What value did you specify for the RootCachePath? It needs to be writable folder by the user.

`C:\Users\Admin\AppData\Roaming\FaceNet\` and `C:\Users\Admin\AppData\Roaming\FaceNet\user1` for the user. It is writable as files are getting created, but 'Cookies' is not changing after, it just stays at 20kb. If I remove the RequestContext from the browser, and leave only the RootPath, it works, but then I don't have isolation.
(Thank you for moving this into discussions, I didn't know about it)

**Comment by owly111** [comment]
I fixed it, my bad. I was initializing the browser before the Cef settings. Thank you for your help @amaitland
